batch/data-plane/src/test/java/com/microsoft/azure/batch/BatchIntegrationTestBase.java

```diff
@@ -1,52 +1,242 @@
-                    System.getenv("CLIENT_ID"), System.getenv("APPLICATION_SECRET"), "microsoft.onmicrosoft.com", null,
-                    null);
+            credentials = getApplicationTokenCredentials();
         } else {
-            credentials = new BatchSharedKeyCredentials(System.getenv("AZURE_BATCH_ENDPOINT"),
-                    System.getenv("AZURE_BATCH_ACCOUNT"), System.getenv("AZURE_BATCH_ACCESS_KEY"));
+            credentials = getSharedKeyCredentials();
         }
         batchClient = BatchClient.open(credentials);
     }
 
+    void createClientWithInterceptor(AuthMode mode) throws IOException {
+        BatchCredentials credentials;
+
+        interceptorManager = InterceptorManager.create(testName.getMethodName(), testMode);
+        RestClient restClient;
+
+        credentials = getCredentials(mode);
+
+        if (isRecordMode()) {
+
+            restClient = buildRestClient(new RestClient.Builder().withBaseUrl(credentials.baseUrl())
+                    .withSerializerAdapter(new AzureJacksonAdapter())
+                    .withCredentials(credentials)
+                    .withResponseBuilderFactory(new ResponseBuilder.Factory() {
+                        private final AzureResponseBuilder.Factory baseFactory = new AzureResponseBuilder.Factory();
+                        @Override
+                        public <T, E extends RestException> ResponseBuilder<T, E> newInstance(SerializerAdapter<?> serializerAdapter) {
+                            return baseFactory.<T, E>newInstance(serializerAdapter).withThrowOnGet404(true);
+                        }
+                    })
+                    .withLogLevel(LogLevel.NONE)
+                    .withNetworkInterceptor(new LoggingInterceptor(LogLevel.BODY_AND_HEADERS))
+                    .withNetworkInterceptor(interceptorManager.initInterceptor())
+                    .withInterceptor(new ResourceManagerThrottlingInterceptor()));
+
+            batchClient = BatchClient.open(restClient, credentials.baseUrl());
+            alternativeBatchClient = batchClient;
+
+        } else { // is Playback Mode
+            out = System.out;
+            System.setOut(new PrintStream(new OutputStream() {
+                public void write(int b) {
+                    // DO NOTHING
+                }
+            }));
+
+            batchClient = BatchClient.open(buildPlaybackRestClient(credentials, playbackUri + "/"),playbackUri+"/");
+            alternativeBatchClient = BatchClient.open(buildPlaybackRestClient(credentials, alternativePlaybackUri + "/"),alternativePlaybackUri+"/");
+
+        }
+    }
+
+    private static BatchCredentials getCredentials(AuthMode mode){
+        BatchCredentials credentials;
+        if(isRecordMode()) {
+            if (mode == AuthMode.AAD) {
+                credentials = getApplicationTokenCredentials();
+            } else {
+                credentials = getSharedKeyCredentials();
+            }
+        } else{
```

Comment:
Have you tested this against CheckStyle? This brace with the missing space should cause an issue (or are all tests ignored)?

Reply:
All tests were ignored previously.
Enabled Test Source directory checkstyle checks in the plugin.
Fixed the checkstyle issues in this class.